solr/core/src/java/org/apache/solr/handler/ReplicationHandler.java

```diff
@@ -1513,257 +1544,11 @@ public void postSoftCommit() {}
     };
   }
 
-  /** This class is used to read and send files in the lucene index */
-  private class DirectoryFileStream implements SolrCore.RawWriter {
```

Comment:
minor: as you are moving things and implementing a new interface, it's an opportunity to consider the ordering of the methods follows a logical flow.  Disclaimer:  I don't know this code enough to know what that is; it may be ideal already.  We like to read top-down generally so if it flows that way then great.